RecoTauTag/RecoTau/interface/PFRecoTauClusterVariables.h

```diff
@@ -22,29 +23,41 @@ class TauIdMVAAuxiliaries {
     /// return chi2 of the leading track ==> deprecated? <==
     float tau_leadTrackChi2(const reco::PFTau& tau) const {
       float LeadingTracknormalizedChi2 = 0;
-      const reco::PFCandidatePtr& leadingPFCharged = tau.leadPFChargedHadrCand() ;
+      const reco::CandidatePtr& leadingPFCharged = tau.leadChargedHadrCand();
       if (leadingPFCharged.isNonnull()) {
-        reco::TrackRef tref = leadingPFCharged -> trackRef();
-        if (tref.isNonnull()) {
-          LeadingTracknormalizedChi2 = (float)(tref -> normalizedChi2());
+      	const reco::PFCandidate* pfcand = dynamic_cast<const reco::PFCandidate*>(leadingPFCharged.get());
+      	if (pfcand != nullptr) {
+          reco::TrackRef tref = pfcand->trackRef();
```

Comment:
I don't know how to avoid the cast here without other structural changes: 
PFTau::leadingPFCharged will throw when there is a leading Candidate that is not a PFCandidate. This throw is the current design (similar to accessing PFCandidates from pat::Tau), which I don't see how to avoid without introducing other, in my opinion less nice solutions like a flag in the PFTau that indicates how the PFTau was made.
I should also point out here that we don't need this variable/method in the latest MVA trainings, so I've also adapted the MVA evaluation so that it only uses this method when evaluating an old MVA training (which shouldn't be the case anymore in future RECOs).

Reply:
I meant to simply change ```const reco::PFCandidate* pfcand = dynamic_cast<const reco::PFCandidate*>(leadingCharged.get())``` to ```const reco::PFCandidate* pfcand = tau.leadingPFCharged.get();```

Reply:
This would throw for PFTaus made from PackedCandidates, since PFTau::leadingPFCharged() will throw an exception (telling that the PFTau was not made PFCandidates). As I tried to describe in the previous comment, I see no nice way around this, and I also tried to explain that I prefer not to change it since the method here is not going to be used anymore, except for reading old MVA ID trainings.

Reply:
ah, you are correct.
ignore my previous comment on this.